astropy/coordinates/builtin_frames/ecliptic_transforms.py

```diff
@@ -26,8 +26,19 @@ def _ecliptic_rotation_matrix(equinox):
     jd1, jd2 = get_jd12(equinox, 'tt')
     rnpb = erfa.pnm06a(jd1, jd2)
     obl = erfa.obl06(jd1, jd2)*u.radian
-    return np.asarray(np.dot(rotation_matrix(obl, 'x'), rnpb))
-
+    """
+    The following code is the equivalent of looping over obl and rnpb,
+    creating a rotation matrix from obl and then taking
+    the dot product of the resulting matrices, finally combining
+    into a new array.
+    """
+    try:
+        rmat = np.array([rotation_matrix(this_obl, 'x') for this_obl in obl])
+        result = np.einsum('...ij,...jk->...ik', rmat, rnpb)
```

Comment:
In numpy >=1.10, this is more efficiently done with `np.matmul`.  Since eventually this should be used, we may as well replace now with (`matmul` should work for scalar case too):
```
from ...utils.compat import NUMPY_LT_1_10

if NUMPY_LT_1_10:
    try:
        result = np.einsum('...ij,...jk->...ik', rmat, rnpb)
    ...
else:
    result = np.matmul(rmat, rnpb)